src/Panel/Panel.tsx

```diff
@@ -1,189 +1,228 @@
+    const [expanded, setExpanded] = useControlled<boolean>(expandedProp, defaultExpanded);
+
+    const handleToggle = useCallback(() => {
+      setExpanded(!expanded);
+    }, [expanded, setExpanded]);
+
+    const handleSelect = useCallback(
+      (event: React.MouseEvent) => {
+        event.persist();
+        onSelect?.(eventKey, event);
+        handleToggle();
+      },
+      [eventKey, handleToggle, onSelect]
     );
-  }
 
-  renderAnchor(header: React.ReactNode, headerRole?: string) {
-    const { id, collapsible } = this.props;
+    const renderAnchor = useCallback(
+      (header: React.ReactNode, headerRole?: string) => (
+        <span
+          aria-controls={collapsible ? `${id}` : null}
+          className={expanded ? null : 'collapsed'}
+          aria-expanded={expanded}
+          aria-selected={expanded}
+          role={headerRole}
+        >
+          {header}
+        </span>
+      ),
+      [collapsible, expanded, id]
+    );
 
-    return (
-      <span
-        aria-controls={collapsible ? `${id}` : null}
-        className={this.isExpanded() ? null : 'collapsed'}
-        aria-expanded={this.isExpanded()}
-        aria-selected={this.isExpanded()}
-        role={headerRole}
-      >
-        {header}
-      </span>
+    const renderCollapsibleTitle = useCallback(
+      (header: React.ReactNode, headerRole?: string) => {
+        return (
+          <span className={prefix('title')} role="presentation">
+            {renderAnchor(header, headerRole)}
+          </span>
+        );
+      },
+      [prefix, renderAnchor]
     );
-  }
 
-  render() {
-    const {
-      headerRole,
-      panelRole,
-      className,
-      collapsible,
-      bordered,
-      shaded,
-      classPrefix,
-      id,
-      ...props
-    } = this.props;
+    const renderBody = useCallback(() => {
+      const classes = merge(prefix('body'), {
+        [prefix('body-fill')]: bodyFill
+      });
+
+      return <div className={classes}>{children}</div>;
+    }, [bodyFill, children, merge, prefix]);
+
+    const renderCollapsibleBody = useCallback(
+      (panelRole?: string) => (
+        <Collapse
+          in={expanded}
+          onEnter={onEnter}
+          onEntering={onEntering}
+          onEntered={onEntered}
+          onExit={onExit}
+          onExiting={onExiting}
+          onExited={onExited}
+        >
+          {(transitionProps, ref) => {
+            const { className, ...rest } = transitionProps;
+            return (
+              <div
+                {...rest}
+                id={id ? `${id}` : null}
+                aria-hidden={!expanded}
+                role={panelRole}
+                className={merge(className, prefix('collapse'))}
+                ref={ref}
+              >
+                {renderBody()}
+              </div>
+            );
+          }}
+        </Collapse>
+      ),
+      [expanded, id, merge, prefix, renderBody]
+    );
```

Comment:
React Hook useCallback has missing dependencies: 'onEnter', 'onEntered', 'onEntering', 'onExit', 'onExited', and 'onExiting'. Either include them or remove the dependency array